File: docs/benchmark/github-source/cases/46-cs-topological-sort/output.cs
using DataStructures.Graph;

namespace Algorithms.Graph;

/// <summary>
///     Topological Sort is a linear ordering of vertices in a Directed Acyclic Graph (DAG)
///     such that for every directed edge (u, v), vertex u comes before vertex v in the ordering.
///
///     KEY CONCEPTS:
///     1. Only applicable to Directed Acyclic Graphs (DAGs) - graphs with no cycles.
///     2. A DAG can have multiple valid topological orderings.
///     3. Used in dependency resolution, task scheduling, build systems, and course prerequisites.
///
///     ALGORITHM APPROACHES:
///     1. DFS-based (Depth-First Search): Uses post-order traversal and reverses the result.
///     2. Kahn's Algorithm: Uses in-degree counting and processes vertices with zero in-degree.
///
///     TIME COMPLEXITY: O(V + E) where V is vertices and E is edges.
///     SPACE COMPLEXITY: O(V) for the visited set and result stack.
///
///     Reference: "Introduction to Algorithms" (CLRS) by Cormen, Leiserson, Rivest, and Stein.
///     Also covered in "Algorithm Design Manual" by Steven Skiena.
/// </summary>
/// <typeparam name="T">Vertex data type.</typeparam>
public class TopologicalSort<T> where T : IComparable<T>
{
    { … 337 line(s) … ⟦tj:c25d9fa7c8a6c92870e3030581976deb⟧ }
[omitted blocks are individually retrievable: call tinyjuice_retrieve with the token inside an omission marker to expand just that block]

[compacted tool output — this is a PARTIAL view; the full original (14419 bytes) is available by calling tinyjuice_retrieve with token "5dc7491667a2d80f9805a7947293b5f7" (marker ⟦tj:5dc7491667a2d80f9805a7947293b5f7⟧)]

File: docs/benchmark/github-source/cases/43-c-trie/output.c
/*------------------Trie Data Structure----------------------------------*/
/*-------------Implimented for search a word in dictionary---------------*/

/*-----character - 97 used for get the character from the ASCII value-----*/

// needed for strnlen
#define _POSIX_C_SOURCE 200809L

#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define ALPHABET_SIZE 26

/*--Node in the Trie--*/
struct trie {
    struct trie *children[ALPHABET_SIZE];
    bool end_of_word;
};


/*--Create new trie node--*/
int trie_new (
    struct trie ** trie
)
{
    { … 7 line(s) … ⟦tj:14b067434c370241c52d7829f2ce6c05⟧ }


/*--Insert new word to Trie--*/
int trie_insert (
    struct trie * trie,
    char *word,
    unsigned word_len
)
{
    { … 38 line(s) … ⟦tj:dc2373e1161e007ee2b6c1b1326f3aea⟧ }


/*--Search a word in the Trie--*/
int trie_search(
    struct trie * trie,
    char *word,
    unsigned word_len,
    struct trie ** result
)
{
    { … 29 line(s) … ⟦tj:b0752dd8150f1f318d7abdcd281b31df⟧ }

/*---Return all the related words------*/
void trie_print (
    struct trie * trie,
    char prefix[],
    unsigned prefix_len
)
{
    { … 24 line(s) … ⟦tj:0d4d1fefe61ad9f6b47286da312ae463⟧ }


/*------Demonstrate purposes uses text file called dictionary -------*/

int main() {
    { … 49 line(s) … ⟦tj:e9fe61775b8c626c074b333e01b4a672⟧ }
[omitted blocks are individually retrievable: call tinyjuice_retrieve with the token inside an omission marker to expand just that block]

[compacted tool output — this is a PARTIAL view; the full original (5241 bytes) is available by calling tinyjuice_retrieve with token "6310a979a27595cc7a51260fd786f142" (marker ⟦tj:6310a979a27595cc7a51260fd786f142⟧)]